modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/SuggestedFoldersManager.kt

```diff
@@ -0,0 +1,45 @@
+package au.com.shiftyjelly.pocketcasts.repositories.podcast
+
+import au.com.shiftyjelly.pocketcasts.models.db.AppDatabase
+import au.com.shiftyjelly.pocketcasts.models.db.dao.SuggestedFoldersDao
+import au.com.shiftyjelly.pocketcasts.models.entity.SuggestedFolder
+import au.com.shiftyjelly.pocketcasts.servers.podcast.PodcastCacheServiceManager
+import au.com.shiftyjelly.pocketcasts.servers.podcast.SuggestedFoldersRequest
+import jakarta.inject.Inject
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import timber.log.Timber
+
+class SuggestedFoldersManager @Inject constructor(
+    private val podcastCacheService: PodcastCacheServiceManager,
+    appDatabase: AppDatabase,
+) : CoroutineScope {
```

Comment:
I think we should avoid extending `CoroutineScope` and inject one if it is needed. I know that we have this pattern in some parts of the code but it is not a good pattern. It hinders testing and doesn't offer much in return. More often then not we don't even need to have a scope and a simple suspending function is enough.